commands/dkcheck.ml

```diff
@@ -1,141 +1,181 @@
-          Arg.Set Pp.print_db_enabled,
-          " Prints De Bruijn indices in error messages" );
-        ( "--confluence",
-          Arg.String Confluence.set_cmd,
-          "CMD Set the external confluence checker command to CMD" );
-        ( "--beautify",
-          Arg.Set beautify,
-          " Pretty printer. Print on the standard output" );
-        ( "--version",
-          Arg.Unit (fun () -> Format.printf "Dedukti %s@." Version.version),
-          " Prints the version number" )
-        (* Deprecated flags. TODO: Remove them from the argument parsing. *);
-        deprecated "-errors-in-snf" "--snf" (Arg.Set Env.errors_in_snf);
-        deprecated "-cc" "--confluence" (Arg.String Confluence.set_cmd);
-        deprecated "-eta" "--eta"
-          (Arg.Tuple [Arg.Set Reduction.eta; Arg.Clear Env.check_arity]);
-        deprecated "-coc" "--coc" (Arg.Set Typing.coc);
-        deprecated "-nl" "no flag" (Arg.Unit ignore);
-        deprecated "-version" "--version"
-          (Arg.Unit (fun () -> Format.printf "Dedukti %s@." Version.version));
-      ]
+  Arg.(value & opt string "" & info ["d"; "debug"] ~docv:"FLAGS" ~doc ~docs)
+
+let quiet =
+  let doc = "Do not print anything (same as -d q)" in
+  let docs = Manpage.s_common_options in
+  Arg.(value & flag & info ["q"; "quiet"] ~doc ~docs)
+
+let verbose =
+  let doc = "Verbose mode (equivalent to -d montru). Overrides -d/--debug." in
+  let docs = Manpage.s_common_options in
+  Arg.(value & flag & info ["v"; "verbose"] ~doc ~docs)
+
+let config_t =
+  let fn no_color debug quiet verbose = {no_color; debug; quiet; verbose} in
+  Term.(const fn $ no_color $ debug $ quiet $ verbose)
+
+let init (c : config) : unit =
+  (* Verbose overrides debug that overrides quiet. *)
+  if c.quiet then Env.set_debug_mode "q";
+  Env.set_debug_mode c.debug;
+  if c.verbose then Env.set_debug_mode "montru";
+  Errors.color := not c.no_color
+
+(** dkcheck options *)
+
+let incl =
+  let doc = "Add directory $(docv) to the load path" in
+  Arg.(value & opt_all dir [] & info ["I"; "include"] ~docv:"DIR" ~doc)
+
+let export =
+  let doc = "Generate object files (\".dko\")" in
+  Arg.(value & flag & info ["e"] ~doc)
+
+let beautify =
+  (* TODO put in a separate command. *)
+  let doc = "Pretty print a file on the standard input" in
+  Arg.(value & flag & info ["beautify"] ~doc)
+
+let run_on_stdin =
+  let doc = "Parses standard input using module name $(docv)" in
+  Arg.(value & opt (some string) None & info ["stdin"] ~docv:"MOD" ~doc)
+
+let de_bruijn =
+  let doc = "Print de Bruijn indices in error messages" in
+  Arg.(value & flag & info ["db"] ~doc)
+
+let confluence =
+  let doc =
+    "Use $(docv) as external confluence checker. $(docv) must be a readable \
+     file."
   in
+  Arg.(
+    value & opt (some string) None & info ["cc"; "confluence"] ~doc ~docv:"CMD")
+
+let files =
+  let doc = "Dedukti files to type check" in
+  Arg.(non_empty & pos_all string [] & info [] ~docv:"FILE" ~doc)
```

Comment:
It seems you are making this argument mandatory while it was not before. Is this intentional?

Reply:
Fixed in b14304a